src/util/csvParser/CsvParser.ts

```diff
@@ -0,0 +1,36 @@
+import { parse } from "csv-parse/sync";
+import { CsvErrorCode } from "./CsvErrors";
+
+export const parseCsv = (data, wantedColumns) => {
+  try {
+    const records = parse(data, {
+      columns: true,
+      skip_empty_lines: true
+    });
+    return records;
+  } catch (err) {
+    if (err.code === "CSV_RECORD_INCONSISTENT_COLUMNS") {
+      if (err.columns.length !== wantedColumns.length) {
+        throw {
+          code: CsvErrorCode.ColumnsLengthMissmatch,
+          expected: wantedColumns,
+          received: err.columns
+        };
+      }
+      for (const column of wantedColumns) {
+        if (!err.columns.some(element => element.name === column)) {
+          throw { code: CsvErrorCode.MissingColumn, column: column };
+        }
+      }
+      if (err.message.includes("Invalid Record Length")) {
```

Comment:
what's an invalid record length? does it have anything to do with the column count?

Reply:
This would be the case when the columns are fine but one line is missing them